src/Compilers/Test/Core/Platform/Custom/MetadataSignatureHelper.cs

```diff
@@ -13,6 +13,10 @@
 using System.Text;
 using Roslyn.Utilities;
 
+// The compiler must emit binary serialization metadata entries even though it doesn't 
+// actually do binary serialization.
+#pragma warning disable SYSLIB0050 
```

Comment:
😨

Reply:
Yeah it's an interesting problem. The technology is being deprecated but compiler still has to emit and read it cause it will live forever in net472

Reply:
I'm also surprised the compiler needs to do special stuff for this.